src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DocumentProcessor.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItemRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.DocumentManagementService;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.output.Pdf;
+
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class DocumentProcessor {
+
+    private final DocumentManagementService documentManagementService;
+    private final ScannableItemRepository scannableItemRepository;
+
+    private static final Logger log = LoggerFactory.getLogger(BlobProcessor.class);
+
+    public DocumentProcessor(
+        DocumentManagementService documentManagementService,
+        ScannableItemRepository scannableItemRepository
+    ) {
+        this.documentManagementService = documentManagementService;
+        this.scannableItemRepository = scannableItemRepository;
+    }
+
+    @Transactional
+    public void processPdfFiles(List<Pdf> pdfs, List<ScannableItem> scannedItems) {
+        Map<String, String> response = documentManagementService.uploadDocuments(pdfs);
+
+        log.info("Document service response with file name and doc url {}", response);
+
+        scannedItems.forEach(item -> {
+            if (response.containsKey(item.getFileName())) {
+                item.setDocumentUrl(response.get(item.getFileName()));
+            } else {
+                //throw some exception here
+                log.error("Document url not found for document {}", item.getFileName());
+            }
+        });
+
+        scannableItemRepository.saveAll(scannedItems);
```

Comment:
i'd rather keep those items as fire and forget as seen throughout various java practices / implementations. any of this typical modification is preferred to have separate private list to which you add elements. could be overthinking and this approach is good enough 👍